app/src/main/java/com/celzero/bravedns/receiver/BraveScreenStateReceiver.kt

```diff
@@ -32,19 +34,19 @@ class BraveScreenStateReceiver : BroadcastReceiver() {
     override fun onReceive(context: Context?, intent: Intent?) {
         if (intent!!.action.equals(Intent.ACTION_SCREEN_OFF)) {
             if(DEBUG) Log.d(LOG_TAG,"BraveScreenStateReceiver : Action_screen_off detected from the receiver")
-            if(PersistentState.getFirewallModeForScreenState(context!!) && !PersistentState.getScreenLockData(context)) {
+            if(ReceiverHelper.persistentState.getFirewallModeForScreenState() && !ReceiverHelper.persistentState.getScreenLockData()) {
                 if(DEBUG) Log.d(LOG_TAG,"BraveSCreenStateReceiver : Screen lock data not true, calling DeviceLockService service")
                 val newIntent = Intent(context, DeviceLockService::class.java)
                 newIntent.action = DeviceLockService.ACTION_CHECK_LOCK
                 newIntent.putExtra(DeviceLockService.EXTRA_STATE, intent.action)
-                context.startService(newIntent)
+                context!!.startService(newIntent)
             }
         } else if (intent.action.equals(Intent.ACTION_USER_PRESENT) || intent.action.equals(Intent.ACTION_SCREEN_ON)) {
             if(DEBUG) Log.d(LOG_TAG,"BraveScreenStateReceiver : ACTION_USER_PRESENT/ACTION_SCREEN_ON detected from the receiver")
-            if(PersistentState.getFirewallModeForScreenState(context!!)) {
-                val state = PersistentState.getScreenLockData(context)
+            if(ReceiverHelper.persistentState.getFirewallModeForScreenState()) {
```

Comment:
Is `ReceiverHelper` here a static singleton or an instance of type `ReceiverHelper`?

Reply:
It is a Kotlin `object`, which is a singleton. It inherits from `KoinComponent` to be able to use `inject()` and provide the required services.